### PR TITLE
Add ORF Radio Stations

### LINKED ISSUE #24938: Please add support for https://noe.orf.at/
## Checklist

- [x] I'm reporting a new site support request
- [x] I've verified that I'm running youtube-dl version **2020.03.24**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that none of provided URLs violate any copyrights
- [x] I've searched the bugtracker for similar site support requests including closed ones

## Example URLs

- Single video: https://noe.orf.at/player/20200419/NN/1587280144000
- Single video: https://noe.orf.at/player/20200419/NN/1587280585000
- Playlist: https://noe.orf.at/player/20200419/NN

## Description

This is a radio station often featuring interesting interviews and stories I wish to store. Please consider adding support.

### PR DESCRIPTION
**Pull request by hh0rva1h**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [x] New feature

---

### Description of your *pull request* and other information

This adds support for the various ORF radio stations. Tested with the following links:
- https://noe.orf.at/player/20200423/NGM/1587610800000
- https://wien.orf.at/player/20200423/WGUM/1587610793000
- https://burgenland.orf.at/player/20200423/BGM/1587611000000
- https://ooe.orf.at/player/20200423/OGMO
- https://steiermark.orf.at/player/20200423/STGMS
- https://kaernten.orf.at/player/20200423/KGUMO
- https://salzburg.orf.at/player/20200423/SGUM
- https://tirol.orf.at/player/20200423/TGUMO
- https://vorarlberg.orf.at/player/20200423/VGUM

Closes #24938

### CODE REVIEW

File: youtube_dl/extractor/orf.py [hunk 1]
```diff
@@ -166,9 +166,40 @@ def _real_extract(self, url):
...
+            loop_station = "oe2s"
+        elif station == "tirol":
+            api_station = "tir"
+            loop_station = "oe2t"
+        elif station == "vorarlberg":
+            api_station = "vbg"
+            loop_station = "oe2v"
+        else:
+            loop_station = station
+            api_station = station
```
Comment (dstftw) [inline youtube_dl/extractor/orf.py:198]: Move these to individual extractors.

File: youtube_dl/extractor/orf.py [hunk 2]
```diff
@@ -223,6 +254,66 @@ class ORFFM4IE(ORFRadioIE):
     }
 
 
+class ORFNOEIE(ORFRadioIE):
+    IE_NAME = 'orf:noe'
+    IE_DESC = 'Radio Niederösterreich'
+    _VALID_URL = r'https?://(?P<station>noe)\.orf\.at/player/(?P<date>[0-9]+)/(?P<show>\w+)'
```
Comment (dstftw) [inline youtube_dl/extractor/orf.py:260]: Add `only_matching` test for each extractor.

Comment (hh0rva1h) [reply youtube_dl/extractor/orf.py:260]: Are md5 and info_dict required? All the shows wont be available after 7 days ...
Is the md5 the md5sum of the downloaded file?

Comment (dstftw) [reply youtube_dl/extractor/orf.py:260]: No, only_matching as already pointed out.

Comment (hh0rva1h) [reply youtube_dl/extractor/orf.py:260]: Also there will be multiple files downloaded in general ...

Comment (dstftw) [reply youtube_dl/extractor/orf.py:231]: Again: **ONLY MATCHING** test, no downloading, no metadata, nothing.

Comment (hh0rva1h) [reply youtube_dl/extractor/orf.py:231]: Oh ok, so not even the url field is supposed to be there?

Comment (hh0rva1h) [reply youtube_dl/extractor/orf.py:231]: Sry I saw your comments too late.

File: youtube_dl/extractor/orf.py [hunk 3]
```diff
@@ -280,14 +309,24 @@ class ORFSBGIE(ORFRadioIE):
...
+        'only_matching': True,
+    }
+
 
 class ORFTIRIE(ORFRadioIE):
-    IE_NAME = 'orf:salzburg'
-    IE_DESC = 'Radio Salzburg'
+    IE_NAME = 'orf:tirol'
+    IE_DESC = 'Radio Tirol'
     _VALID_URL = r'https?://(?P<station>salzburg)\.orf\.at/player/(?P<date>[0-9]+)/(?P<show>\w+)'
```
Comment (lcnittl) [inline youtube_dl/extractor/orf.py:314]: Shouldn't here be Tirol in the URL?

```
_VALID_URL = r'https?://(?P<station>tirol)\.orf\.at/player/(?P<date>[0-9]+)/(?P<show>\w+)'
```

Comment (hh0rva1h) [reply youtube_dl/extractor/orf.py]: True, thanks for catching that. I had tirol duplicated two times and removed the right duplicate unfortunately. Is fixed now.

**Comment by hh0rva1h** [comment]
@dstftw Should be ready for merge.

File: youtube_dl/extractor/orf.py [hunk 4]
```diff
@@ -205,6 +204,8 @@ class ORFFM4IE(ORFRadioIE):
     IE_NAME = 'orf:fm4'
     IE_DESC = 'radio FM4'
     _VALID_URL = r'https?://(?P<station>fm4)\.orf\.at/player/(?P<date>[0-9]+)/(?P<show>4\w+)'
+    api_station = "fm4"
+    loop_station = "fm4"
```
Comment (dstftw) [inline youtube_dl/extractor/orf.py:208]: Const must be uppercase. Protected must be prefix with underscore. Single quotes.

Comment (hh0rva1h) [reply youtube_dl/extractor/orf.py]: Thanks very much, fixed.

**Pull request merged**